presto-main/src/main/java/com/facebook/presto/sql/planner/RelationPlanner.java

```diff
@@ -294,6 +295,7 @@ else if (firstDependencies.stream().allMatch(right.canResolvePredicate()) && sec
 
         PlanNode root = new JoinNode(idAllocator.getNextId(),
                 JoinNode.Type.typeConvert(node.getType()),
+                DISTRIBUTED,
```

Comment:
why distributed is default? should not be take from some session property?

See https://prestodb.io/docs/current/admin/tuning.html and `distributed-joins-enabled`

Then you would not need to use `Session` in the optimizer.